packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx

```diff
@@ -237,6 +247,40 @@ export const OAuthLoginPage = () => {
     dispatch(signOut())
   }
 
+  const { data: currentWeb3User } = useGetCurrentWeb3User({})
+  const { data: currentUserId } = useGetCurrentUserId({})
+  const { switchAccount } = useAccountSwitcher()
+
+  const onAccountSelected = useCallback(
+    (user: UserMetadata) => {
+      switchAccount(user)
+    },
+    [switchAccount]
+  )
+
+  const web3UserId = currentWeb3User?.user_id ?? null
+
+  const { data: managedAccounts = [] } = useGetManagedAccounts(
+    { userId: web3UserId! },
+    { disabled: !web3UserId }
+  )
+
+  const accounts = useMemo(() => {
+    return managedAccounts.filter(({ grant }) => grant.is_approved)
+  }, [managedAccounts])
+
+  const isInManagerMode = account?.user_id === currentWeb3User?.user_id
+
+  const [isAccountSwitcherOpen, setAccountSwitcherOpen] = useState(false)
+
+  const onOpenAccountSwitcher = useCallback(() => {
+    setAccountSwitcherOpen(true)
+  }, [setAccountSwitcherOpen])
+
+  const onCloseAccountSwitcher = useCallback(() => {
```

Comment:
Bleh I feel like this is way too much required boilerplate 😅 
I maybe consider moving the whole state management for the switcher into a single hook the way we do for modals.

Reply:
I don't think it's all that bad! it took like 30 seconds to copy it all correctly and it also helped build my understanding of what the data is, but I'm down to consolidate it into one hook too